src/main/java/com/varchenko/MikeHashMap.java

```diff
@@ -0,0 +1,282 @@
+package com.varchenko;
+
+
+import java.util.*;
+
+// Node -> null, Node -> key null, Node -> key, value
+public class MikeHashMap<K, V> extends AbstractMap<K, V> implements Map<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] buckets;
+    private float loadFactor;
+    private int capacity;
+    private int modCount;
+    private int size;
+
+    public MikeHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        buckets = (Node<K, V>[]) new Node[capacity];
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+    }
+
+    public MikeHashMap(int capacity, float loadFactor) {
+        this();
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+    }
+
+    private void resize() {
+        if (size >= capacity * loadFactor) {
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[capacity + (capacity >> 1)];
```

Comment:
The value of capacity doesn't change. It's a problem.

Reply:
done